lib/dao.js

```diff
@@ -1908,8 +1929,9 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
   var idNames = Model.definition.idNames();
   for (var i = 0, n = idNames.length; i < n; i++) {
     var idName = idNames[i];
-    if (data[idName] !== undefined && data[idName] !== inst[idName]) {
-      var err = new Error('id property value cannot be updated: ' + idName);
+    if (data[idName] !== undefined && !idEquals(data[idName], inst[idName])) {
+      var err = new Error('id property (' + idName +
+        ') cannot be updated from ' + inst[idName] + ' to ' + data[idName]);
```

Comment:
Nitpick: The formatting is a bit strange, seeing a closing brace at the start of a line feels weird: